app/views/AttachmentView/CachedVideo.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { useEffect, useState, useRef } from 'react';
+import { ActivityIndicator, View, StyleSheet, StyleProp, ViewStyle } from 'react-native';
+import { Video, ResizeMode } from 'expo-av';
+import * as FileSystem from 'expo-file-system';
+
+interface CachedVideoProps {
+	videoUrl: string;
+	fallbackOnError?: () => void;
+	style?: StyleProp<ViewStyle>;
+	resizeMode?: ResizeMode;
+	shouldPlay?: boolean;
+	isLooping?: boolean;
+	setLoading?: React.Dispatch<React.SetStateAction<boolean>>;
+	useNativeControls?: boolean;
+	loading?: boolean;
+}
+
+const CachedVideo: React.FC<CachedVideoProps> = ({
+	videoUrl,
+	fallbackOnError = () => {},
+	style,
+	resizeMode = ResizeMode.CONTAIN,
+	shouldPlay = true,
+	isLooping = false,
+	useNativeControls = true,
+	setLoading = () => {},
+	loading = false,
+	...props
+}) => {
+	const [localUri, setLocalUri] = useState<string | null>(null);
+	const videoRef = useRef<Video>(null);
+
+	useEffect(() => {
+		const cacheVideo = async () => {
+			try {
+				const fileName = videoUrl.split('/').pop() || `cached-video-${Date.now()}.mp4`;
+				const localPath = FileSystem.documentDirectory + fileName;
+				const fileInfo = await FileSystem.getInfoAsync(localPath);
+
+				if (!fileInfo.exists) {
+					await FileSystem.downloadAsync(videoUrl, localPath);
```

Comment:
You shouldn't have to do this.
Downloads should be handled here already https://github.com/RocketChat/Rocket.Chat.ReactNative/blob/1327f588f4b3735a36deed9aafe55b12eeece85c/app/lib/methods/handleMediaDownload.ts#L275
Your PR should fix it there, if there is a bug.